### DISCUSSION TITLE
Versioning Strategy

### DISCUSSION THREAD

**Discussion by CPretorius1986** [microsoft/AL-Go#429] [Q&A] [answered]
Good day Everyone,

I am trying to get to the bottom of versioning in AL GO,

I have read the ALGo setting for versioning strategy but only see 3 and not very clear on further options.

If we wish to use the version that is set in the app? how would we go about it? i.e. override the "auto" versioning and use the version stated in the app.json

Thanks.

**Comment by freddydk** [comment, marked as answer]
I will add some more documentation on versioning.
I did have a question raised here, which has a rather detailed answer:
https://github.com/microsoft/AL-Go/issues/421

We always use major.minor from app.json and always calculate build.release.
Currently there is no way to just use the version number in app.json as you should never ever have to different app files with the same version number - this will lead to problems down the road.

**Comment by freddydk** [comment]
After this - we added a new versioning strategy = 3, where the 3 first digits are taken from app.json and repoVersion - and only the last digit is defined by gitHub.
